### PR TITLE
Create Document for Custom User Model

### PR DESCRIPTION
**Pull request by ZhangChen199102**
Created ADR for the custom User model introduced by #47

### CODE REVIEW

File: docs/decisions/0003-custom-user-model.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+[The official Django documentation][document] highly recommends using a custom user model when
+start a Django project, because it’s hard to switch a user model out for another once
+the project is deployed and being used by real users,
+
+## Implications
+
+In some cases, users might lost access to their email address before they update their
+email address on the website or APP, and if the project uses email as primary user identifier
+it will be difficult for them to get their accounts back.
```
Comment (a-musing-moose) [inline docs/decisions/0003-custom-user-model.md:30]: **suggestion:** I would also add a paragraph here saying that we realise this the build user model we provide here might not be _perfect_ for all use cases, but by making a custom user model part of the template project we can insure it is as easy as possible to change and evolve that model later one. Something this is a lot more difficult should we stick with the built in one.‏

File: docs/decisions/0003-custom-user-model.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# 0002 Use Custom User Model
+
+- Date: 2024-04-10
+- Author(s): [Chen Zhang][chen]
+- Status: `Active`
+
+## Decision
+
+We will use a custom User model instead of using Django built-in User model.
```
Comment (a-musing-moose) [inline docs/decisions/0003-custom-user-model.md:9]: **issue:** ‏Can we also state here that the custom model will be _based_ on the built in one and provide all the same features?

**Review by a-musing-moose — CHANGES_REQUESTED** [review]
This fine - a couple of extra paragraph to give a little more information about you have been doing with the custom user model would, I think, help clarify things.

I have added suggestions in line.

File: docs/decisions/0003-custom-user-model.md (hunk 2 above)
Comment (ZhangChen199102) [reply docs/decisions/0003-custom-user-model.md:9]: Thanks, this makes lot of sense.

File: docs/decisions/0003-custom-user-model.md (hunk 1 above)
Comment (ZhangChen199102) [reply docs/decisions/0003-custom-user-model.md:30]: Thanks, this makes lot of sense.

File: docs/decisions/0003-custom-user-model.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+[The official Django documentation][document] highly recommends using a custom user
+model when start a Django project, because it’s hard to switch a user model out for
+another once the project is deployed and being used by real users.
+
+## Implications
+
+This custom User model might not be perfect for all use cases,  but by making a custom
+user model part of the template project we can insure it is as easy as possible to
+change and evolve that model later one. Something this is a lot more difficult should
+we stick with the built in one.
```
Comment (a-musing-moose) [inline docs/decisions/0003-custom-user-model.md:33]: **typo:** ‏

```suggestion
change and evolve that model later one. Something that is a lot more difficult should
we stick with the built in one.
```

Sorry, this one is my fault.

**Review by a-musing-moose — APPROVED** [review]

**Pull request merged**